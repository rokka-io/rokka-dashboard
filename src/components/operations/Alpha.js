import PropTypes from 'prop-types'
import React, { Component } from 'react'
import Select from '../forms/Select'
import FormGroup from '../forms/FormGroup'

class Alpha extends Component {
  isRequired (field) {
    return this.props.required.indexOf(field) !== -1
  }

  render () {
    const { defaults, values, errors = {} } = this.props

    console.log(defaults, values, errors)
    const $alphaModeOptions = defaults.mode.values
      .map((option) => {
        return (
          <option key={option} value={option}>
            {option}
          </option>
        )
      })

    return (
      <div className="row">
        <div className="col-md-6">
          <FormGroup label="Alpha mode" required={this.isRequired('alpha')}
            error={errors.mode}>
            <Select
              name="mode"
              className="rka-select"
              value={values.mode}
              defaultValue={defaults.mode.default}
              onChange={this.props.onChange}>
              {$alphaModeOptions}
            </Select>
          </FormGroup>
        </div>
      </div>
    )
  }
}

Alpha.propTypes = {
  onChange: PropTypes.func,
  defaults: PropTypes.shape({
    mode: PropTypes.shape({
      values: PropTypes.array
    })
  }),
  values: PropTypes.shape({
    rotation_direction: PropTypes.string
  }),
  required: PropTypes.array.isRequired,
  errors: PropTypes.object
}

export default Alpha
