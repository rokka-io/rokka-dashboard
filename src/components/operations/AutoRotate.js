import PropTypes from 'prop-types'
import React, { PureComponent } from 'react'
import Input from '../forms/Input'
import Select from '../forms/Select'
import FormGroup from '../forms/FormGroup'

class AutoRotate extends PureComponent {
  isRequired(field) {
    return this.props.required.indexOf(field) !== -1
  }

  render() {
    const { defaults, values, errors = {} } = this.props

    const $rotationDirectionOptions = defaults.rotation_direction.values.map(option => {
      return (
        <option key={option} value={option}>
          {option}
        </option>
      )
    })

    return (
      <div className="row">
        <div className="col-md-4">
          <FormGroup label="Width" required={this.isRequired('width')} error={errors.width}>
            <Input
              name="width"
              type="text"
              className="rka-input-txt"
              placeholder="width"
              value={values.width}
              defaultValue={defaults.width.default}
              onChange={this.props.onChange}
            />
          </FormGroup>
        </div>
        <div className="col-md-4">
          <FormGroup label="Height" required={this.isRequired('height')} error={errors.height}>
            <Input
              name="height"
              type="text"
              className="rka-input-txt"
              placeholder="height"
              value={values.height}
              defaultValue={defaults.height.default}
              onChange={this.props.onChange}
            />
          </FormGroup>
        </div>
        <div className="col-md-4">
          <FormGroup
            label="Rotation direction"
            required={this.isRequired('rotation_direction')}
            error={errors.rotation_direction}
          >
            <Select
              name="rotation_direction"
              className="rka-select"
              value={values.rotation_direction}
              defaultValue={defaults.rotation_direction.default}
              onChange={this.props.onChange}
            >
              {$rotationDirectionOptions}
            </Select>
          </FormGroup>
        </div>
      </div>
    )
  }
}

AutoRotate.propTypes = {
  onChange: PropTypes.func,
  defaults: PropTypes.shape({
    width: PropTypes.shape({
      min: PropTypes.number,
      max: PropTypes.number
    }),
    height: PropTypes.shape({
      min: PropTypes.number,
      max: PropTypes.number
    }),
    rotation_direction: PropTypes.shape({
      values: PropTypes.array
    })
  }),
  values: PropTypes.shape({
    height: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    rotation_direction: PropTypes.string
  }),
  required: PropTypes.array.isRequired,
  errors: PropTypes.object
}

export default AutoRotate
