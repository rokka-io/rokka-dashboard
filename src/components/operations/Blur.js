import PropTypes from 'prop-types'
import React, { Component } from 'react'
import InputRange from '../forms/InputRange.js'
import FormGroup from '../forms/FormGroup'

class Blur extends Component {
  isRequired (field) {
    return this.props.required.indexOf(field) !== -1
  }

  render () {
    const { defaults, values, errors = {} } = this.props

    return (
      <div className="row">
        <div className="col-md-6">
          <FormGroup label="Blur Sigma" required={this.isRequired('sigma')} error={errors.sigma}>
            <InputRange
              name="sigma"
              min={defaults.sigma.minimum}
              max={defaults.sigma.maximum}
              value={values.sigma}
              defaultValue={defaults.sigma.default}
              onChange={this.props.onChange}
            />
          </FormGroup>
        </div>
        <div className="col-md-6">
          <FormGroup label="Blur Radius" required={this.isRequired('radius')} error={errors.radius}>
            <InputRange
              name="radius"
              min={defaults.radius.minimum}
              max={defaults.radius.maximum}
              value={values.radius}
              defaultValue={defaults.radius.default}
              onChange={this.props.onChange}
            />
          </FormGroup>
        </div>
      </div>
    )
  }
}

Blur.propTypes = {
  onChange: PropTypes.func,
  defaults: PropTypes.shape({
    radius: PropTypes.shape({
      minimum: PropTypes.number,
      maximum: PropTypes.number
    }),
    sigma: PropTypes.shape({
      minimum: PropTypes.number,
      maximum: PropTypes.number
    })
  }),
  values: PropTypes.shape({
    radius: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    sigma: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
  }),
  required: PropTypes.array.isRequired,
  errors: PropTypes.object
}

export default Blur
