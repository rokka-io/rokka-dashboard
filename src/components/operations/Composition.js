import React, { PureComponent, PropTypes } from 'react'
import Input from '../forms/Input'
import InputColor from '../forms/InputColor'
import Select from '../forms/Select'
import InputRange from '../forms/InputRange'
import FormGroup from '../forms/FormGroup'

class Composition extends PureComponent {
  isRequired (field) {
    return this.props.required.indexOf(field) !== -1
  }

  render () {
    const { defaults, values, errors = {} } = this.props

    return (
      <div className="row">
        <div className="col-md-6">
          <FormGroup label="Width" required={this.isRequired('width')} error={errors.width}>
            <InputRange name="width" min={defaults.width.minimum} max={defaults.width.maximum}
              value={values.width} defaultValue={defaults.width.default} onChange={this.props.onChange} />
          </FormGroup>
        </div>
        <div className="col-md-6">
          <FormGroup label="Height" required={this.isRequired('height')} error={errors.height}>
            <InputRange name="height" min={defaults.height.minimum} max={defaults.height.maximum}
              value={values.height} defaultValue={defaults.height.default} onChange={this.props.onChange} />
          </FormGroup>
        </div>
        <div className="col-md-6">
          <FormGroup label="Mode" required={this.isRequired('mode')} error={errors.mode}>
            <Select
              name="mode"
              className="rka-select"
              value={values.mode}
              defaultValue={defaults.mode.default}
              onChange={this.props.onChange}>
              {defaults.mode.values.map(option => <option key={option} value={option}>{option}</option>)}
            </Select>
          </FormGroup>
        </div>
        <div className="col-md-6">
          <FormGroup label="Anchor" required={this.isRequired('anchor')} error={errors.anchor}>
            <Input
              name="anchor"
              type="text"
              className="rka-input-txt"
              placeholder="Anchor"
              defaultValue={defaults.anchor.default}
              value={values.anchor}
              onChange={this.props.onChange}
            />
          </FormGroup>
        </div>
        <div className="col-md-6">
          <FormGroup label="Secondary Color" required={this.isRequired('secondary_color')} error={errors.color}>
            <InputColor name="secondary_color" value={values.secondary_color || defaults.secondary_color.default}
              onChange={this.props.onChange} />
          </FormGroup>
        </div>
        <div className="col-md-6">
          <FormGroup label="Secondary opacity" required={this.isRequired('secondary_opacity')} error={errors.secondary_opacity}>
            <InputRange name="secondary_opacity" min={defaults.secondary_opacity.minimum} max={defaults.secondary_opacity.maximum}
              value={values.secondary_opacity} defaultValue={defaults.secondary_opacity.default} onChange={this.props.onChange} />
          </FormGroup>
        </div>
      </div>
    )
  }
}

Composition.propTypes = {
  onChange: PropTypes.func,
  defaults: PropTypes.shape({
    mode: PropTypes.shape({
      values: PropTypes.array
    }),
    width: PropTypes.shape({
      minimum: PropTypes.number,
      maximum: PropTypes.number
    }),
    height: PropTypes.shape({
      minimum: PropTypes.number,
      maximum: PropTypes.number
    }),
    anchor: PropTypes.shape({
      default: PropTypes.string
    }),
    secondary_color: PropTypes.shape({
      default: PropTypes.string
    }),
    secondary_opacity: PropTypes.shape({
      minimum: PropTypes.number,
      maximum: PropTypes.number
    })
  }),
  values: PropTypes.shape({
    mode: PropTypes.string,
    width: PropTypes.string,
    height: PropTypes.string,
    anchor: PropTypes.string,
    secondary_color: PropTypes.string,
    secondary_opacity: PropTypes.string
  }),
  required: PropTypes.array.isRequired,
  errors: PropTypes.object
}

export default Composition
