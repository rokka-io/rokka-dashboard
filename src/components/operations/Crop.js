import PropTypes from 'prop-types'
import React, { Component } from 'react'
import Input from '../forms/Input'
import InputRange from '../forms/InputRange'
import FormGroup from '../forms/FormGroup'

class Crop extends Component {
  isRequired (field) {
    return this.props.required.indexOf(field) !== -1
  }

  render () {
    const { defaults, values, errors = {} } = this.props

    return (
      <div className="row">
        <div className="col-md-6">
          <FormGroup label="Width" required={this.isRequired('width')} error={errors.width}>
            <InputRange name="width" min={defaults.width.minimum} max={defaults.width.maximum}
              value={values.width} defaultValue={defaults.width.value} onChange={this.props.onChange} />
          </FormGroup>
        </div>
        <div className="col-md-6">
          <FormGroup label="Height" required={this.isRequired('height')} error={errors.height}>
            <InputRange name="height" min={defaults.height.minimum} max={defaults.height.maximum}
              value={values.height} defaultValue={defaults.height.value} onChange={this.props.onChange} />
          </FormGroup>
        </div>
        <div className="col-md-6">
          <FormGroup label="Anchor" required={this.isRequired('anchor')} error={errors.anchor}>
            <Input
              name="anchor"
              type="text"
              className="rka-input-txt"
              placeholder="Anchor"
              defaultValue={defaults.anchor.default}
              value={values.anchor}
              onChange={this.props.onChange}
            />
          </FormGroup>
        </div>
      </div>
    )
  }
}

Crop.propTypes = {
  onChange: PropTypes.func,
  defaults: PropTypes.shape({
    width: PropTypes.shape({
      minimum: PropTypes.number,
      maximum: PropTypes.number
    }),
    height: PropTypes.shape({
      minimum: PropTypes.number,
      maximum: PropTypes.number
    })
  }),
  values: PropTypes.shape({
    width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    height: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    anchor: PropTypes.string
  }),
  required: PropTypes.array.isRequired,
  errors: PropTypes.object
}

export default Crop
