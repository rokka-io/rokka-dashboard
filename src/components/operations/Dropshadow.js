import PropTypes from 'prop-types'
import React, { Component } from 'react'
import InputRange from '../forms/InputRange'
import InputColor from '../forms/InputColor'
import FormGroup from '../forms/FormGroup'

class Dropshadow extends Component {
  isRequired (field) {
    return this.props.required.indexOf(field) !== -1
  }

  render () {
    const { defaults, values, errors = {} } = this.props

    return (
      <div className="row">
        <div className="col-md-4">
          <FormGroup label="Horizontal" required={this.isRequired('horizontal')} error={errors.horizontal}>
            <InputRange name="horizontal" min={defaults.horizontal.minimum} max={defaults.horizontal.maximum}
              value={values.horizontal} defaultValue={defaults.horizontal.default} onChange={this.props.onChange} />
          </FormGroup>
        </div>
        <div className="col-md-4">
          <FormGroup label="Vertical" required={this.isRequired('vertical')} error={errors.vertical}>
            <InputRange name="vertical" min={defaults.vertical.minimum} max={defaults.vertical.maximum}
              value={values.vertical} defaultValue={defaults.vertical.default} onChange={this.props.onChange} />
          </FormGroup>
        </div>
        <div className="col-md-4">
          <FormGroup label="Opacity" required={this.isRequired('opacity')} error={errors.opacity}>
            <InputRange name="opacity" min={defaults.opacity.minimum} max={defaults.opacity.maximum}
              value={values.opacity} defaultValue={defaults.opacity.default} onChange={this.props.onChange} />
          </FormGroup>
        </div>
        <div className="col-md-4">
          <FormGroup label="Color" required={this.isRequired('color')} error={errors.color}>
            <InputColor name="color" value={values.color} defaultValue={defaults.color.default} onChange={this.props.onChange} />
          </FormGroup>
        </div>
        <div className="col-md-4">
          <FormGroup label="Blur radius" required={this.isRequired('blur_radius')} error={errors.blur_radius}>
            <InputRange name="blur_radius" min={defaults.blur_radius.minimum} max={defaults.blur_radius.maximum}
              value={values.blur_radius} defaultValue={defaults.blur_radius.default} onChange={this.props.onChange} />
          </FormGroup>
        </div>
        <div className="col-md-4">
          <FormGroup label="Sigma" required={this.isRequired('sigma')} error={errors.sigma}>
            <InputRange name="sigma" min={defaults.sigma.minimum} max={defaults.sigma.maximum}
              value={values.sigma} defaultValue={defaults.sigma.default} onChange={this.props.onChange} />
          </FormGroup>
        </div>
      </div>
    )
  }
}

Dropshadow.propTypes = {
  onChange: PropTypes.func,
  defaults: PropTypes.shape({
    horizontal: PropTypes.shape({
      minimum: PropTypes.number,
      maximum: PropTypes.number
    }),
    vertical: PropTypes.shape({
      minimum: PropTypes.number,
      maximum: PropTypes.number
    }),
    opacity: PropTypes.shape({
      minimum: PropTypes.number,
      maximum: PropTypes.number
    }),
    sigma: PropTypes.shape({
      minimum: PropTypes.number,
      maximum: PropTypes.number
    }),
    blur_radius: PropTypes.shape({
      minimum: PropTypes.number,
      maximum: PropTypes.number
    })
  }),
  values: PropTypes.shape({
    horizontal: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    vertical: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    opacity: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    sigma: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    blur_radius: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
  }),
  required: PropTypes.array.isRequired,
  errors: PropTypes.object
}

export default Dropshadow
