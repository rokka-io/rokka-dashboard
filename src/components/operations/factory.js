import deepAssign from 'deep-assign'
import React, { createElement } from 'react'
import Alpha from './Alpha'
import AutoRotate from './AutoRotate'
import Blur from './Blur'
import Composition from './Composition'
import Crop from './Crop'
import Dropshadow from './Dropshadow'
import Grayscale from './Grayscale'
import Noop from './Noop'
import Primitive from './Primitive'
import Resize from './Resize'
import Rotate from './Rotate'
import Sepia from './Sepia'
import Trim from './Trim'

const operations = {
  alpha: Alpha,
  autorotate: AutoRotate,
  blur: Blur,
  composition: Composition,
  crop: Crop,
  dropshadow: Dropshadow,
  grayscale: Grayscale,
  noop: Noop,
  primitive: Primitive,
  resize: Resize,
  rotate: Rotate,
  sepia: Sepia,
  trim: Trim
}

const defaults = {
  resize: {
    upscale: {
      values: ['true', 'false']
    },
    upscale_dpr: {
      values: ['true', 'false']
    }
  },
  rotate: {
    angle: {
      default: 0
    }
  },
  dropshadow: {
    blur_radius: {
      maximum: 10000
    }
  }
}

/**
 * Creates the correct component based on the name. E.g. resize => Resize component.
 *
 * @param {object}   availableOperations
 * @param {string}   name
 * @param {*}        values
 * @param {Function} onChange
 * @param {Object}    errors
 */
export default function (availableOperations, name, values, onChange = null, errors = {}) {
  if (operations[name]) {
    const props = {
      defaults: deepAssign(availableOperations[name].properties, defaults[name]),
      required: availableOperations[name].required || [],
      values,
      onChange,
      errors: errors
    }

    return createElement(operations[name], props)
  }

  const issueSearch = encodeURIComponent(`is:issue ${name}`)

  return (
    <div className="row">
      <div className="rka-alert is-info">
        Stack operation <span className="txt-bold">{name}</span> not implemented yet in rokka-dashboard.
        {' '}
        <a className="rka-link txt-white" target="_blank" href={`https://github.com/rokka-io/rokka-dashboard/issues?q=${issueSearch}`}>File an issue on GitHub</a>
      </div>
    </div>
  )
}
