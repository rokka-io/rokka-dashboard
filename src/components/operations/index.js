import PropTypes from 'prop-types'
import React from 'react'
import { findDOMNode } from 'react-dom'
import { DragSource, DropTarget } from 'react-dnd'
import cx from 'classnames'
import factory from './factory'

const ITEM_TYPE = 'OPERATION'

const Operation = ({
  availableOperations,
  operation,
  index,
  isActive,
  onChange,
  removeOperation,
  setActiveOperation,
  isDragging,
  isOver,
  connectDragSource,
  connectDragPreview,
  connectDropTarget
}) => {
  const hasErrors = Object.keys(operation.errors).length
  const className = cx('pa-md', 'mb-xs', 'pos-r', {
    'bg-gray-lightest': index % 2,
    'is-active': isActive || hasErrors,
    'bor-cranberry': hasErrors,
    'bor-light': !hasErrors,
    'opaque': isDragging,
    'bor-brand-light-dashed': isOver
  })

  return connectDragPreview(connectDropTarget(
    <div className={className} key={`operations-${index}`}>
      {connectDragSource(
        <div>
          <a className="rka-close-icon" href="#" onClick={e => removeOperation(e, index)} title="Remove" />
          <a className="rka-edit-icon" href="#" onClick={e => setActiveOperation(e, index)} />
          <a className="rka-move-icon" href="#" onClick={e => e.preventDefault()} />
          <h3 className="rka-h3 txt-cap">{operation.name}</h3>
        </div>
      )}
      <div className={cx({'semi-transparent': isOver})}>
        <div className="rka-operation-content">
          {factory(availableOperations, operation.name, operation.options, (e) => onChange(index, e), operation.errors)}
        </div>
      </div>
    </div>
  ))
}
Operation.propTypes = {
  availableOperations: PropTypes.object.isRequired,
  operation: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    options: PropTypes.object.isRequired,
    errors: PropTypes.object.isRequired
  }).isRequired,
  index: PropTypes.number.isRequired,
  isActive: PropTypes.bool.isRequired,
  // state manipulators on container
  onChange: PropTypes.func.isRequired,
  onMoveOperation: PropTypes.func.isRequired,
  removeOperation: PropTypes.func.isRequired,
  setActiveOperation: PropTypes.func.isRequired,
  // from HOCs of react-dnd
  connectDragSource: PropTypes.func.isRequired,
  connectDragPreview: PropTypes.func.isRequired,
  connectDropTarget: PropTypes.func.isRequired,
  isDragging: PropTypes.bool.isRequired,
  isOver: PropTypes.bool.isRequired
}

const dragSourceSpec = {
  beginDrag (props) {
    return {
      id: props.operation.id,
      index: props.index
    }
  },

  endDrag (props, monitor) {
    const { id: droppedId, originalIndex } = monitor.getItem()
    const didDrop = monitor.didDrop()

    if (!didDrop) {
      props.onMoveOperation(droppedId, originalIndex)
    }
  }
}

const dragSourceCollect = (connect, monitor) => ({
  connectDragSource: connect.dragSource(),
  connectDragPreview: connect.dragPreview(),
  isDragging: monitor.isDragging()
})

const dropTargetSpec = {
  hover (props, monitor, component) {
    const dragIndex = monitor.getItem().index
    const hoverIndex = props.index

    // Don't replace items with themselves
    if (dragIndex === hoverIndex) {
      return
    }

    // Determine rectangle on screen
    const hoverBoundingRect = findDOMNode(component).getBoundingClientRect()

    // Get vertical middle
    const hoverMiddleY = (hoverBoundingRect.bottom - hoverBoundingRect.top) / 2

    // Determine mouse position
    const clientOffset = monitor.getClientOffset()

    // Get pixels to the top
    const hoverClientY = clientOffset.y - hoverBoundingRect.top

    // Only perform the move when the mouse has crossed half of the items height
    // When dragging downwards, only move when the cursor is below 50%
    // When dragging upwards, only move when the cursor is above 50%

    // Dragging downwards
    if (dragIndex < hoverIndex && hoverClientY < hoverMiddleY) {
      return
    }

    // Dragging upwards
    if (dragIndex > hoverIndex && hoverClientY > hoverMiddleY) {
      return
    }

    // Time to actually perform the action
    props.onMoveOperation(dragIndex, hoverIndex)

    // Note: we're mutating the monitor item here!
    // Generally it's better to avoid mutations,
    // but it's good here for the sake of performance
    // to avoid expensive index searches.
    monitor.getItem().index = hoverIndex
  }
}

const dropTargetCollect = (connect, monitor) => ({
  connectDropTarget: connect.dropTarget(),
  isOver: monitor.isOver()
})

export default DropTarget(ITEM_TYPE, dropTargetSpec, dropTargetCollect)(DragSource(ITEM_TYPE, dragSourceSpec, dragSourceCollect)(Operation))
