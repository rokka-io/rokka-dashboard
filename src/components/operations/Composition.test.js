import React from 'react'
import Composition from '../../../src/components/operations/Composition'
import renderer from 'react-test-renderer'
import operations from '../../__tests__/operations.json'

const compositionValues = {
  mode: 'foreground',
  width: 1000,
  height: 1000,
  anchor: 'center',
  secondary_color: '222222',
  primary_color: '111111'
}

test('Composition does render', () => {
  const component = renderer.create(
    <Composition values={compositionValues} defaults={operations.composition.properties} required={[]} />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
