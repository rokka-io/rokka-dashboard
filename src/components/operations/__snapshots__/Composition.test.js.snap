// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Composition does render 1`] = `
<div
  className="row"
>
  <div
    className="col-md-6"
  >
    <div
      className="rka-form-group"
    >
      <label
        className="rka-label"
        htmlFor={null}
      >
        Width
      </label>
      <div
        className="rka-input-range-container clearfix"
      >
        <input
          className="rka-input-range"
          disabled={true}
          max={10000}
          min={1}
          name="width"
          onBlur={[Function]}
          onChange={[Function]}
          type="range"
          value={1000}
        />
        <p>
          <span
            className="rka-input-range-min"
          >
            1
          </span>
          <span
            className="rka-input-range-current txt-bold is-disabled"
          >
            1000
          </span>
          <span
            className="rka-input-range-max"
          >
            10000
          </span>
        </p>
      </div>
    </div>
  </div>
  <div
    className="col-md-6"
  >
    <div
      className="rka-form-group"
    >
      <label
        className="rka-label"
        htmlFor={null}
      >
        Height
      </label>
      <div
        className="rka-input-range-container clearfix"
      >
        <input
          className="rka-input-range"
          disabled={true}
          max={10000}
          min={1}
          name="height"
          onBlur={[Function]}
          onChange={[Function]}
          type="range"
          value={1000}
        />
        <p>
          <span
            className="rka-input-range-min"
          >
            1
          </span>
          <span
            className="rka-input-range-current txt-bold is-disabled"
          >
            1000
          </span>
          <span
            className="rka-input-range-max"
          >
            10000
          </span>
        </p>
      </div>
    </div>
  </div>
  <div
    className="col-md-6"
  >
    <div
      className="rka-form-group"
    >
      <label
        className="rka-label"
        htmlFor={null}
      >
        Mode
      </label>
      <select
        className="rka-select"
        disabled={true}
        name="mode"
        value="foreground"
      >
        <option />
        <option
          value="foreground"
        >
          foreground
        </option>
        <option
          value="background"
        >
          background
        </option>
      </select>
    </div>
  </div>
  <div
    className="col-md-6"
  >
    <div
      className="rka-form-group"
    >
      <label
        className="rka-label"
        htmlFor={null}
      >
        Resize mode
      </label>
      <select
        className="rka-select"
        disabled={true}
        name="resize_mode"
        placeholder="resize_mode (default: box)"
        value="box"
      >
        <option>
          resize_mode (default: box)
        </option>
        <option
          value="box"
        >
          box
        </option>
        <option
          value="absolute"
        >
          absolute
        </option>
        <option
          value="fill"
        >
          fill
        </option>
      </select>
    </div>
  </div>
  <div
    className="col-md-6"
  >
    <div
      className="rka-form-group"
    >
      <label
        className="rka-label"
        htmlFor={null}
      >
        Anchor
      </label>
      <input
        className="rka-input-txt"
        disabled={true}
        name="anchor"
        placeholder="Anchor (default: auto)"
        type="text"
        value="center"
      />
    </div>
  </div>
  <div
    className="col-md-6"
  >
    <div
      className="rka-form-group"
    >
      <label
        className="rka-label"
        htmlFor={null}
      >
        Secondary Color
      </label>
      <div
        className="rka-form-group"
      >
        <input
          className="rka-input-txt"
          disabled={true}
          name="secondary_color"
          onClick={[Function]}
          placeholder="color (default: 000000)"
          readOnly={true}
          type="text"
          value="222222"
        />
      </div>
    </div>
  </div>
  <div
    className="col-md-6"
  >
    <div
      className="rka-form-group"
    >
      <label
        className="rka-label"
        htmlFor={null}
      >
        Secondary opacity
      </label>
      <div
        className="rka-input-range-container clearfix"
      >
        <input
          className="rka-input-range"
          disabled={true}
          max={100}
          min={0}
          name="secondary_opacity"
          onBlur={[Function]}
          onChange={[Function]}
          type="range"
          value={100}
        />
        <p>
          <span
            className="rka-input-range-min"
          >
            0
          </span>
          <span
            className="rka-input-range-current txt-bold is-disabled"
          >
            100
          </span>
          <span
            className="rka-input-range-max"
          >
            100
          </span>
        </p>
      </div>
    </div>
  </div>
  <div
    className="col-md-6"
  >
    <div
      className="rka-form-group"
    >
      <label
        className="rka-label"
        htmlFor={null}
      >
        Secondary image (hash)
      </label>
      <input
        className="rka-input-txt"
        disabled={true}
        name="secondary_image"
        placeholder="Secondary image (hash)"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
